frontend/packages/core/src/AppProvider/tests/registrar.test.tsx

```diff
@@ -34,86 +35,198 @@ const WORKFLOW_CONFIG = {
-    delete gatewayCfg["@clutch-sh/ec2"].terminateInstance.componentProps;
-    workflowRoutes(WORKFLOW_ID, WORKFLOW_CONFIG, gatewayCfg);
+  it("warns on invalid gateway route configurations", () => {
+    delete gatewayRoutes.terminateInstance.componentProps;
+    workflowRoutes(WORKFLOW_ID, WORKFLOW_CONFIG, gatewayRoutes);
     expect(warn).toHaveBeenCalledWith(
-      "[@clutch-sh/ec2][instance/terminate] Not registered: Invalid config - missing required component props resolverType"
+      "[@clutch-sh/ec2][terminateInstance] Invalid gateway config: route is missing required props"
     );
   });
-  it("filters out routes missing required props", () => {
-    const gatewayCfg = { ...USER_CONFIGURATION };
-    delete gatewayCfg["@clutch-sh/ec2"].terminateInstance.componentProps;
-    const routes = workflowRoutes(WORKFLOW_ID, WORKFLOW_CONFIG, gatewayCfg);
+
+  it("removes routes with invalid gateway configurations", () => {
+    delete gatewayRoutes.terminateInstance.componentProps;
+    const routes = workflowRoutes(WORKFLOW_ID, WORKFLOW_CONFIG, gatewayRoutes);
     expect(routes).toHaveLength(0);
   });
 
-  it("returns valid routes", () => {
-    const routes = workflowRoutes(WORKFLOW_ID, WORKFLOW_CONFIG, USER_CONFIGURATION);
+  it("removes empty routes", () => {
+    // @ts-ignore
+    const routes = workflowRoutes(WORKFLOW_ID, WORKFLOW_CONFIG, { terminateInstance: {} });
     expect(routes).toEqual([]);
   });
+
+  it("returns valid routes", () => {
+    const routes = workflowRoutes(WORKFLOW_ID, WORKFLOW_CONFIG, gatewayRoutes);
+    expect(routes).toHaveLength(1);
+  });
+});
+
+describe("registeredWorkflows", () => {
+  let warn;
+  let gatewayConfig;
+  const availableWorkflows = { [WORKFLOW_ID]: () => WORKFLOW_CONFIG };
+  beforeEach(() => {
+    gatewayConfig = JSON.parse(JSON.stringify(GATEWAY_CONFIG));
+    warn = jest.spyOn(console, "warn").mockImplementation(() => {});
+  });
+
+  afterEach(() => {
+    warn.mockClear();
+  });
+
+  afterAll(() => {
+    warn.mockRestore();
+  });
+
+  it("handles empty workflows", () => {
+    return registeredWorkflows().then(workflows => {
+      expect(workflows).toHaveLength(0);
+    });
+  });
+
+  it("handles empty gateway configuration", () => {
+    return registeredWorkflows(undefined).then(workflows => {
+      expect(workflows).toHaveLength(0);
+    });
+  });
+
+  it("allows workflow overrides from gateway", () => {
+    gatewayConfig[WORKFLOW_ID].overrides = { displayName: "Amazon Web Services" };
+    return registeredWorkflows(availableWorkflows, gatewayConfig).then(workflows => {
+      expect(workflows[0].displayName).toBe("Amazon Web Services");
+    });
+  });
+
+  it("removes workflow overrides from gateway config", () => {
+    gatewayConfig[WORKFLOW_ID].overrides = { displayName: "Amazon Web Services" };
+    return registeredWorkflows(availableWorkflows, gatewayConfig).then(() => {
+      expect(warn).not.toHaveBeenCalled();
+    });
+  });
```

Comment:
is there a typo in the test description? not sure what is getting "removed"

Reply:
oh i think i understand now https://github.com/lyft/clutch/pull/1741/files#diff-7157db281f494cb33c18ac118bfd99f9ec25a1645e1e9097b15f07e808a825a8R95